src/main/java/io/pinecone/PineconeClient.java

```diff
@@ -36,4 +36,9 @@ public PineconeConnection connect(String indexName) {
     public PineconeConnection connect(PineconeConnectionConfig connectionConfig) {
         return new PineconeConnection(config, connectionConfig);
     }
+
+    public PineconeConnection connectWithUrl(String connectionUrl) {
```

Comment:
Approach 1: 
connectWithUrl(String apiKey, String connectionUrl) {
// Initialize clientConfig.withApiKey()
// Initialize connection.withConnectionUrl()
}

Approach 1 requires user to enter api key for every new indexes which doesn't make sense since apiKey doesnt change for each connectionUrl/indexes. Rather, use the existing clientConfig with apiKey and initialize a new PineconeConnection object which internally initializes PineconeConfigClient with connectionUrl. Please let me know your thoughts on this if you've a better approach or different opinion.

Reply:
Agreed, this feels like it makes sense here. 

**nit:** Also wondering if we should be adding a little documentation type comment here similar to the other `connect()` methods.